lib/rubocop/cop/style/if_unless_modifier.rb

```diff
@@ -20,15 +20,38 @@ def on_if(node)
           return if modifier_if?(node)
           return if elsif?(node)
           return if if_else?(node)
-          # Accept cases that require parentheses around modifier if statement.
           return if chained?(node)
           return unless fit_within_line_as_modifier_form?(node)
           add_offense(node, :keyword, message(node.loc.keyword.source))
         end
 
         def chained?(node)
-          ancestor = node.ancestors.first
-          ancestor && ancestor.send_type?
+          # Don't register offense for `if ... end.method`
+          return false if node.parent.nil? || !node.parent.send_type?
+          receiver = node.parent.children[0]
+          node.equal?(receiver)
+        end
+
+        def parenthesize?(node)
+          # Parenthesize corrected expression if changing to modifier-if form
+          # would change the meaning of the parent expression
+          # (due to the low operator precedence of modifier-if)
+          return false if node.parent.nil?
+
+          assignment = [:lvasgn, :casgn, :cvasgn, :gvasgn, :ivasgn, :masgn]
```

Comment:
Will do. BTW, I have sent a patch to `astrolabe` which adds `#assignment_type?` to `Astrolabe::Node`. If they accept, I will rewrite this code to use it.